sample_project/Assets/SampleViewer/Samples/Geometry/Geometries.cs

```diff
@@ -0,0 +1,501 @@
+					}
+					if (featurePoints.Count >= 2)
+						RenderLine(ref featurePoints);
+					RebaseLine();
+				}
+			}
+		}
+
+	}
+
+	private void CreateAndCalculateEnvelope(ArcGISPoint start, ArcGISPoint end)
+	{
+		var spatialReference = new ArcGISSpatialReference(3857);
+
+		var southwestPoint = new ArcGISPoint(Math.Min(start.X, end.X), Math.Min(start.Y, end.Y), spatialReference);
+		var northeastPoint = new ArcGISPoint(Math.Max(start.X, end.X), Math.Max(start.Y, end.Y), spatialReference);
+
+		var envelope = new ArcGISEnvelope(southwestPoint, northeastPoint);
+
+		VisualizeEnvelope(envelope);
+
+		geodeticAreaEnvelope = ArcGISGeometryEngine.AreaGeodetic(envelope, currentAreaUnit, ArcGISGeodeticCurveType.Geodesic);
+		UpdateDisplay();
+	}
+
+	private void VisualizeEnvelope(ArcGISEnvelope envelope)
+	{
+		var bottomLeft = arcGISMapComponent.GeographicToEngine(new ArcGISPoint(envelope.XMin, envelope.YMin, envelope.SpatialReference));
+		var bottomRight = arcGISMapComponent.GeographicToEngine(new ArcGISPoint(envelope.XMax, envelope.YMin, envelope.SpatialReference));
+		var topLeft = arcGISMapComponent.GeographicToEngine(new ArcGISPoint(envelope.XMin, envelope.YMax, envelope.SpatialReference));
+		var topRight = arcGISMapComponent.GeographicToEngine(new ArcGISPoint(envelope.XMax, envelope.YMax, envelope.SpatialReference));
+
+		bottomLeft.y += MarkerHeight;
+		bottomRight.y += MarkerHeight;
+		topLeft.y += MarkerHeight;
+		topRight.y += MarkerHeight;
+
+		var bottomLeftMarker = Instantiate(LineMarker, bottomLeft, Quaternion.identity, arcGISMapComponent.transform);
+		var bottomRightMarker = Instantiate(LineMarker, bottomRight, Quaternion.identity, arcGISMapComponent.transform);
+		var topLeftMarker = Instantiate(LineMarker, topLeft, Quaternion.identity, arcGISMapComponent.transform);
+		var topRightMarker = Instantiate(LineMarker, topRight, Quaternion.identity, arcGISMapComponent.transform);
+
+		SetSurfacePlacement(bottomLeftMarker, MarkerHeight);
+		SetSurfacePlacement(bottomRightMarker, MarkerHeight);
+		SetSurfacePlacement(topLeftMarker, MarkerHeight);
+		SetSurfacePlacement(topRightMarker, MarkerHeight);
+
+		SetElevation(bottomLeftMarker);
+		SetElevation(bottomRightMarker);
+		SetElevation(topLeftMarker);
+		SetElevation(topRightMarker);
+
+		stops.Push(topLeftMarker);
+		stops.Push(topRightMarker);
+		stops.Push(bottomRightMarker);
+		stops.Push(bottomLeftMarker);
+
+		//add feature points in order so that line renderer can correctly connects points; Add start point at the end to close envelope
+		featurePoints.Add(topLeftMarker);
+		Interpolate(topLeftMarker, topRightMarker, featurePoints);
+		featurePoints.Add(topRightMarker);
+		Interpolate(topRightMarker, bottomRightMarker, featurePoints);
+		featurePoints.Add(bottomRightMarker);
+		Interpolate(bottomRightMarker, bottomLeftMarker, featurePoints);
+		featurePoints.Add(bottomLeftMarker);
+		Interpolate(bottomLeftMarker, topLeftMarker, featurePoints);
+		featurePoints.Add(topLeftMarker);
```

Comment:
IS there a reason for having the in-between points? Measuring elevation maybe?

Reply:
The reason for interpolation was 1. to make distance measurement more accurate by accounting for terrains, Basically the calculating was done by summing up all the interpolation intervals. 2. For visual purpose, if there's no interpolation points, the line connecting two markers will go through the terrains.